lib/score.rb

```diff
@@ -0,0 +1,68 @@
+require_relative 'frame'
+
+class Score
+  attr_reader :completed_frames, :current_frame
+
+  def initialize
+    @completed_frames = []
+    @current_frame = Frame.new
+  end
+
+  def check_current_frame
+    if @current_frame.rolls == 2 || @current_frame.strike? || @completed_frames.length >= 10
+      @completed_frames << @current_frame.contents
+      @current_frame = Frame.new
+    end
+    @current_frame.contents
+  end
+
+  def record_roll(roll)
+    roll_limiter
+    @current_frame.record_roll(roll)
+    check_current_frame
+  end
+
+  def calculate_score
+    bonus_roll_correction(apply_bonuses).flatten.sum
+  end
+
+  def apply_bonuses(frames = @completed_frames)
+    frames.each_with_index do |frame, index|
+      if spare?(frame) && (frames[index + 1])
+        frame[-1] += frames[index + 1][0]
+      elsif strike?(frame) && (frames[index + 1]) && (frames[index + 1].length == 2)
+        frame[-1] += (frames[index + 1][0] + frames[index + 1][1])
+      elsif strike?(frame) && (frames[index + 1]) && (frames[index + 2])
+        frame[-1] += (frames[index + 1][0] + frames[index + 2][0])
+      end
+    end
```

Comment:
Did you try using the spare? and strike? methods from the frame class in here rather than making new ones? I feel like there must be a way to just use one set of them, either in here or in the other class. Maybe if you passed @current_frame.contents into the methods below as an argument?